app/addons/dataimporter/components.react.jsx

```diff
@@ -0,0 +1,797 @@
+        <div className="top-row">
+          <div className="big-file-info-message">
+            <p className="big-file-preview-limit-info-message">
+              Because of the size of this file, this preview only shows the
+              first {rowShown} rows, out of {totalRows} rows total.
+              However, if you choose to load the data into a database, the
+              entirety of the file (all {totalRows} rows) will be imported.
+            </p>
+          </div>
+          {this.loadingIntoDB()}
+        </div>
+      );
+    },
+    componentDidMount: function () {
+      document.getElementById('preview-page')
+        .addEventListener('scroll', this.handleScroll);
+    },
+    componentWillUnmount: function () {
+      document.getElementById('preview-page')
+        .removeEventListener('scroll', this.handleScroll);
+    },
+    handleScroll: function (e) {
+      this.setState({left: document.getElementById('preview-page').scrollLeft});
+    },
+
+    loadingIntoDB: function () {
+      if (this.props.isLoadingInDBInProgress) {
+        return (
+          <div className="dataIsLoading">
+            <Components.LoadLines />
+          </div>
+        );
+      }
+      return null;
+    },
+
+    fileMetadataInfo: function () {
+      var totalRows = this.props.rowsTotal,
+          previewMsg = totalRows > 500 ? 'This is a 500 row (max) preview of the file.' : '';
+
+      return (
+        <div className="top-row">
+          <div className="big-file-info-message">
+            <p className="big-file-preview-limit-info-message">
+              All {totalRows} rows from this file will be imported. {previewMsg}
+            </p>
+          </div>
+          {this.loadingIntoDB()}
+        </div>
+      );
+    },
+    render: function () {
+
+      var fileInfoMessage =
+            this.props.isBigFile ? this.bigFilePreviewWarning() :  this.fileMetadataInfo(),
+          style = {'left': this.state.left};
+
+      return (
+        <div id="preview-page" >
+          <div id="data-import-options" style={style}>
+            {fileInfoMessage}
+            <OptionsRow 
+              getDelimiterChosen={this.props.getDelimiterChosen}
+              filesize={this.props.filesize} />
+          </div>
+          <div className="preview-data-space">
+            <TableView
+              data={this.props.data}
+              isBigFile={this.props.isBigFile}
+              meta={this.props.meta}
+              getPreviewView={this.props.getPreviewView}
+              getSmallPreviewOfData={this.props.getSmallPreviewOfData}
+              getHeaderConfig={this.props.getHeaderConfig} />
+            <JSONView
+              data={this.props.data}
+              isBigFile={this.props.isBigFile}
+              meta={this.props.meta}
+              getPreviewView={this.props.getPreviewView}
+              getSmallPreviewOfData={this.props.getSmallPreviewOfData}
+              getHeaderConfig={this.props.getHeaderConfig} />
```

Comment:
together with putting the decision if you use small preview data `getSmallPreviewOfData` (see other comment) into the store you get quite logic less components that just render your state